src/modules/party/party.controller.ts

```diff
@@ -17,19 +29,59 @@ export class PartyController {
   })
   @ApiOkResponse({
     description: 'The matching parties have been successfully retrieved.',
-    type: GetPartiesBySearchTextResponseElement,
+    type: GetPartiesBySearchTextResponseItem,
     isArray: true,
   })
   @ApiInternalServerErrorResponse({
     description: 'An internal server error has occurred.',
   })
-  async getPartiesBySearchText(@Query() query: PartiesQueryDto): Promise<GetPartiesBySearchTextResponse> {
+  async getPartiesBySearchText(@Query() query: GetPartiesBySearchTextQuery): Promise<GetPartiesBySearchTextResponse> {
     const token = await this.acbsAuthenticationService.getIdToken();
     const response = await this.partyService.getPartiesBySearchText(token, query.searchText);
 
     return response;
   }
 
+  @Post()
+  @ApiOperation({
+    summary: 'Post a new party.',
+  })
+  @ApiBody({
+    type: PostPartyRequestItem,
+    isArray: true,
+  })
+  @ApiOkResponse({
+    description: 'One or more parties matching this alternateIdentifier already exist',
+    type: PostPartyResponse,
+  })
+  @ApiCreatedResponse({
+    description: 'Party successfully created.',
+    type: PostPartyResponse,
+  })
+  @ApiBadRequestResponse({
+    description: 'Bad request.',
+  })
+  @ApiInternalServerErrorResponse({
+    description: 'An internal server error has occurred.',
+  })
+  async postParty(
+    @Body(new ParseArrayPipe({ items: PostPartyRequestItem }))
+    requestBody: PostPartyRequest,
+    @Res({ passthrough: true }) res: Response,
+  ): Promise<PostPartyResponse> {
+    const token = await this.acbsAuthenticationService.getIdToken();
+    const response = await this.partyService.getPartyIdentifierBySearchText(token, requestBody[0].alternateIdentifier).then(async (partyIdentifierObj) => {
+      if (partyIdentifierObj) {
+        res.status(HttpStatus.OK);
+        return partyIdentifierObj;
+      } else {
+        return await this.partyService.postParty(token, requestBody[0]);
```

Comment:
We use `requestBody[0]` twice in this method, can we please add a variable for it?